lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -199,9 +199,9 @@ public function setFetchMode($fetchMode, $arg2 = null, $arg3 = null)
      */
     public function getIterator()
     {
-        $data = $this->fetchAll();
-
-        return new \ArrayIterator($data);
+        while ($row = $this->fetch()) {
```

Comment:
I'd replace `$row` with something more generic like `$result`. Otherwise, `$row` creates a false impression that it's always an array, and an empty value means the end of the result set.